homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,253 @@
+            FAN_MEDIUM,
+            FAN_HIGH,
+        ]
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._appliance.nickname
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 60
+        else:
+            return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 90
+        else:
+            return 32
+
+    @property
+    def unique_id(self):
+        """Return unique ID based on Frigidaire ID."""
+        return self._appliance.appliance_id
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        temperature = int(temperature)
+
+        self._client.execute_action(
+            self._appliance, frigidaire.Action.set_temperature(temperature)
+        )
+
+    def set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        convert = {
+            FAN_AUTO: frigidaire.FanSpeed.AUTO,
+            FAN_LOW: frigidaire.FanSpeed.LOW,
+            FAN_MEDIUM: frigidaire.FanSpeed.MEDIUM,
+            FAN_HIGH: frigidaire.FanSpeed.HIGH,
+        }
+
+        # Guard against unexpected fan modes
+        if fan_mode not in convert:
+            return
+
+        action = frigidaire.Action.set_fan_speed(convert[fan_mode])
+        self._client.execute_action(self._appliance, action)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target operation mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            self._client.execute_action(
+                self._appliance, frigidaire.Action.set_power(frigidaire.Power.OFF)
+            )
+            return
+
+        convert = {
+            HVAC_MODE_AUTO: frigidaire.Mode.ECO,
+            HVAC_MODE_FAN_ONLY: frigidaire.Mode.FAN,
+            HVAC_MODE_COOL: frigidaire.Mode.COOL,
+        }
+
+        # Guard against unexpected hvac modes
+        if hvac_mode not in convert:
+            return
+
+        # Turn on if not currently on.
+        if self._details.for_code(frigidaire.HaclCode.AC_MODE) == 0:
+            self._client.execute_action(
+                self._appliance, frigidaire.Action.set_power(frigidaire.Power.ON)
+            )
+
+        self._client.execute_action(
+            self._appliance, frigidaire.Action.set_mode(convert[hvac_mode])
+        )
+
+    def update(self):
+        """Retrieve latest state and updates the details."""
+        try:
+            details = self._client.get_appliance_details(self._appliance)
+            self._details = details
+            self._available = True
+        except (frigidaire.FrigidaireException):
+            _LOGGER.error("Failed to connect to Frigidaire servers")
```

Comment:
To prevent is constantly showing an error in the logs on each try, you could add a check if it was marked unavailable already.

```suggestion
            if self.available:
                _LOGGER.error("Failed to connect to Frigidaire servers")
```